src/Illuminate/Routing/RouteRegistrar.php

```diff
@@ -93,7 +93,14 @@ public function attribute($key, $value)
             throw new InvalidArgumentException("Attribute [{$key}] does not exist.");
         }
 
-        $this->attributes[Arr::get($this->aliases, $key, $key)] = $value;
+        $key = Arr::get($this->aliases, $key, $key);
+
+        if ($key === 'middleware' && isset($this->attributes[$key])) {
+            $this->attributes[$key] = array_merge($this->attributes[$key], $value);
+            return $this;
+        }
+
```

Comment:
Or use
```php
if ($key === 'middleware'){
    $this->attributes['middleware'] = array_merge($this->attributes['middleware'] ?? [], $value);
    return $this;
}
```

Reply:
I've no strong convictions about using $key here or 'middleware', for me — both options is fine. But since we've already checked that $key equals 'middleware', why not use it. And that reduces using string constant (yeah, I understand that we'll never change 'middleware' to something else 😊).
One more argument, in the `__call` method we do check for 'middleware' too, and using `$method` variable in next line. 
https://github.com/laravel/framework/blob/f3f1b059489a5c7a165a5f84d16b81335168b7c3/src/Illuminate/Routing/RouteRegistrar.php#L222